### PR TITLE
DAOS-2877 Docs: Add TransportConfig documentation and update deployme…

### PR DESCRIPTION
**Pull request by dpquigl**
…nt document

When the initial certificate code was merged some copyright notices and code
documentation was overlooked. This adds that documentation and reflects those
changes in the deployment document.

Signed-off-by: David Quigley <david.quigley@intel.com>

### CODE REVIEW

File: doc/admin/deployment.md [hunk 1]
```diff
@@ -1121,7 +1126,10 @@ nodes before starting it.
 
 The DAOS security framework relies on certificates to authenticate
 administrators. The security infrastructure is currently under
-development and will be delivered in DAOS v1.0. Initial support for certificates has been added to DAOS and can be disable either via the command line or in the DAOS Agent configuration file.
+development and will be delivered in DAOS v1.0. Initial support for certificates
+has been added to DAOS and can be disabled either via the command line or in the
+DAOS Agent configuration file. Currently the easiest way to disable certificate
+support is to pass the -i flag to daos_agent.
```
Comment (kjacque) [inline doc/admin/deployment.md]: Need to escape the underscore in daos_agent?

Comment (dpquigl) [reply doc/admin/deployment.md]: Good catch. Probably need to do the same for daos_server further up in the document.

File: doc/admin/deployment.md [hunk 2]
```diff
@@ -1238,7 +1253,8 @@ System Validation
 -----------------
 
 To validate that the DAOS system is properly installed, the daos\_test
-suite can be executed. Ensure the DAOS Agent is configuerd and running before running daos\_test:
+suite can be executed. Ensure the DAOS Agent is configuered and running before
```
Comment (mjmac) [inline doc/admin/deployment.md]: Typo: configuered -> configured

File: doc/admin/deployment.md [hunk 3]
```diff
@@ -1227,7 +1238,11 @@ log\_file: /tmp/daos\_agent2.log
 ## Agent Startup
 --------------
 
-DAOS Agent is a standalone application to be run on each compute node. It can be configured to use secure communications (default) or can be allowed to communicate with the control plane over unencrypted channels. The example below for executing daos_agent specifies to operate in insecure mode as certificate support is not fully integrated into DAOS as of 0.6
+DAOS Agent is a standalone application to be run on each compute node.
+It can be configured to use secure communications (default) or can be allowed
+to communicate with the control plane over unencrypted channels. The example
+below for executing daos_agent specifies to operate in insecure mode as
+certificate support is not fully integrated into DAOS as of 0.6
```
Comment (mjmac) [inline doc/admin/deployment.md]: Missing a trailing period to end the sentence. Also the phrasing is a little awkward. Maybe, "The following example shows daos_agent being configured to operate in insecure mode due to incomplete integration of certificate support as of the 0.6 release." ?

File: doc/admin/deployment.md [hunk 4]
```diff
@@ -1187,7 +1195,9 @@ port: 10001
 
 **Transport Certificate Credentials**
 
-Certificate support for securing the administrative channel for the DAOS components is specified in a key called transport_config: It contains the following keys with these default values
+Certificate support for securing the administrative channel for the DAOS
+components is specified in a key called transport_config: It contains the
+following keys with these default values
```
Comment (mjmac) [inline doc/admin/deployment.md]: Missing trailing period.

**Pull request merged**